### PR TITLE
KOGITO-5894: Fix spotbugs issues

### PR DESCRIPTION
**Pull request by jomarko**

### CODE REVIEW

**Review by domhanak — APPROVED** [review]
LGTM

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/Shape.java [hunk 1]
```diff
@@ -149,15 +150,15 @@ protected void setShapeType(final ShapeType type) {
     @Override
     public Map<ControlHandleType, IControlHandleList> getControlHandles(List<ControlHandleType> types) {
         if ((null == types) || (types.isEmpty())) {
-            return null;
+            return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/Shape.java:153]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java [hunk 2]
```diff
@@ -411,15 +412,15 @@ public C setDragConstraints(final DragConstraintEnforcer enforcer) {
     @Override
     public Map<ControlHandleType, IControlHandleList> getControlHandles(List<ControlHandleType> types) {
         if ((null == types) || (types.isEmpty())) {
-            return null;
+            return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java:415]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java [hunk 3]
```diff
@@ -411,15 +412,15 @@ public C setDragConstraints(final DragConstraintEnforcer enforcer) {
...
         }
         if (types.size() > 1) {
             types = new ArrayList<>(new HashSet<>(types));
         }
         IControlHandleFactory factory = getControlHandleFactory();
 
         if (null == factory) {
-            return null;
+            return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java:423]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java [hunk 4]
```diff
@@ -167,15 +168,15 @@ public C moveToBottom() {
...
+            return new HashMap<>();
         }
         if (types.size() > 1) {
             types = new ArrayList<ControlHandleType>(new HashSet<ControlHandleType>(types));
         }
         IControlHandleFactory factory = getControlHandleFactory();
 
         if (null == factory) {
-            return null;
+            return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java:179]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java [hunk 5]
```diff
@@ -167,15 +168,15 @@ public C moveToBottom() {
     @Override
     public Map<ControlHandleType, IControlHandleList> getControlHandles(List<ControlHandleType> types) {
         if ((null == types) || (types.isEmpty())) {
-            return null;
+            return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java:171]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java [hunk 6]
```diff
@@ -175,10 +175,10 @@ private DefaultMultiPointShapeHandleFactory(final AbstractMultiPointShape<?> sha
         @Override
         public Map<ControlHandleType, IControlHandleList> getControlHandles(final List<ControlHandleType> types) {
             if ((null == types) || (types.isEmpty())) {
-                return null;
+                return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java:178]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java [hunk 7]
```diff
@@ -175,10 +175,10 @@ private DefaultMultiPointShapeHandleFactory(final AbstractMultiPointShape<?> sha
...
             }
             if (false == types.contains(ControlHandleStandardType.POINT) && false == types.contains(ControlHandleStandardType.HANDLE)) {
-                return null;
+                return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java:181]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPathPartShape.java [hunk 8]
```diff
@@ -380,7 +380,7 @@ public DefaultMultiPathShapeHandleFactory(NFastArrayList<PathPartList> listOfPat
         @Override
         public Map<ControlHandleType, IControlHandleList> getControlHandles(List<ControlHandleType> types) {
             if ((null == types) || (types.isEmpty())) {
-                return null;
+                return new HashMap<>();
```
Comment (hasys) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPathPartShape.java:383]: Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

**Review by hasys — COMMENTED** [review]
Good job, thank you @jomarko!

see my comment with question inline

**Review by yesamer — APPROVED** [review]
Thank you, @jomarko
A bit surprised about setters methods wrong behavior like the one present in `Spline.java`.
How it's possible it wasn't caught before? Are these setters really called?

**Comment by jomarko** [comment]
checked `getControlHandles` occurrences
- [ ] DecisionServiceSVGShapeView
- [ ] DecisionServiceSVGShapeViewTest
- [ ] IControlHandleFactory
- [ ] WiresConnector
- [ ] WiresShape
- [ ] AbstractMultiPathPartShape
- [ ] AbstractMultiPointShape
- [ ] CompositeProxy
- [ ] GroupOf
- [ ] Shape
- [ ] WiresShapeTest

**Comment by jomarko** [comment]
@yesamer thank you for your comment https://github.com/kiegroup/kogito-editors-java/pull/106#pullrequestreview-756057231.

Yes it seems most those setters were not used, but `OrthogonalPolyLine.setCornerRadius` seems to be used, so once FDB is working we need to check created vsix. Or should I build it manually and link here guys?

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPathPartShape.java (hunk 8 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPathPartShape.java:383]: agree, good point @hasys , either we stay with `null` or usages of this should be at least checked, and if applies, refactored. I would recommend @jomarko to stay with `null`....

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java (hunk 6 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java:178]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java (hunk 7 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/AbstractMultiPointShape.java:181]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java (hunk 5 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java:171]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java (hunk 4 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java:179]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java (hunk 2 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java:415]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java (hunk 3 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/GroupOf.java:423]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/Shape.java (hunk 1 above)
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/Shape.java:153]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/Shape.java [hunk 9]
```diff
@@ -149,15 +150,15 @@ protected void setShapeType(final ShapeType type) {
...
         }
         if (types.size() > 1) {
             types = new ArrayList<>(new HashSet<>(types));
         }
         IControlHandleFactory factory = getControlHandleFactory();
 
         if (null == factory) {
-            return null;
+            return new HashMap<>();
```
[the remark this answers is not in the document]
Comment (romartin) [reply lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/Shape.java:161]: same here

File: lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/wires/WiresShape.java [hunk 10]
```diff
@@ -222,7 +222,7 @@ private void init() {
     private void _loadControls(final IControlHandle.ControlHandleType type) {
         final Map<ControlHandleType, IControlHandleList> handles = getPath().getControlHandles(type);
 
-        if (null != handles) {
+        if (!handles.isEmpty()) {
```
Comment (romartin) [inline lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/wires/WiresShape.java:225]: if finally staying with the `return null` for the method `getControlHandles` this should be reverted @jomarko , thx!

**Comment by jomarko** [comment]
@romartin I have reverted the `null` -> `new HashMap()` change, please have a look.

**Review by romartin — APPROVED** [review]

**Review by hasys — APPROVED** [review]

**Comment by jomarko** [comment]
@kiegroup/gatekeepers please merge

**Pull request merged**